sandbox/compile_executor.cc

```diff
@@ -0,0 +1,252 @@
+                      .AllowSyscall(__NR_clone)
+                      // [234/tgkill], send a kill signal to a thread. Inparticular used when
+                      // hitting memory limits.
+                      .AllowSyscall(__NR_tgkill)
+                      // Memory management.
+                      .AllowTcMalloc()
+                      // [28/madvise], give advice about use of memory
+                      .AllowSyscall(__NR_madvise)
+                      // [10/mprotect], set protection of a region of memory.
+                      .AllowSyscall(__NR_mprotect)
+                      // [324/membarrier], issue memory barriers.
+                      .AllowSyscall(__NR_membarrier)
+                      // [16/ioctl], used for terminal output.
+                      .AllowSyscall(__NR_ioctl)
+                      // [330/pkey_alloc] V8 uses for querying available memory protection.
+                      .AllowSyscall(__NR_pkey_alloc)
+                      // Needed in v8::base::Stack::GetStackStart().
+                      .AllowSyscall(__NR_sched_getaffinity)
+                      .AllowTime()
+                      .AllowExit()
+                      .AllowGetRandom()
+                      .AllowDynamicStartup()
+                      // For UDS communication.
+                      .AllowSyscall(__NR_rt_sigprocmask)
+                      .AllowSyscall(__NR_rt_sigaction)
+                      .AllowSyscall(__NR_fcntl)
+                      .AllowSyscall(__NR_getsockopt)
+                      .AllowSyscall(__NR_setsockopt)
+                      .AllowSyscall(__NR_sendto)
+                      .AllowSyscall(__NR_shutdown)
+                      .AllowSyscall(__NR_bind)
+                      .AllowSyscall(__NR_listen)
+                      .AllowSyscall(__NR_connect)
+                      .AllowSyscall(__NR_getsockname)
+                      .AllowSyscall(__NR_socket)
+                      .AllowSyscall(__NR_socketpair)
+                      .AllowSyscall(__NR_sendmmsg)
+                      // Allow epoll I/O event notification and piping for fd data transferral.
+                      .AllowSyscall(__NR_epoll_create1)
+                      .AllowSyscall(__NR_epoll_ctl)
+                      .AllowSyscall(__NR_epoll_wait)
+                      .AllowSyscall(__NR_pipe2)
+                      .AllowSyscall(__NR_eventfd2)
+
+                      .AllowSyscall(__NR_clone3)
+                      .AllowSyscall(__NR_sysinfo)
+                      .AllowSyscall(__NR_statx)
+                      .AllowSyscall(__NR_getcwd)
+                      .BuildOrDie();
+
+    sandbox2::Sandbox2 s2(std::move(executor), std::move(policy));
+
+    // If the sandbox program fails to start, return early.
+    if (!s2.RunAsync())
+    {
+        auto result = s2.AwaitResultWithTimeout(
+            absl::Seconds(TIMEOUT_SECS + FALLBACK_TIMEOUT_DELAY));
+        LOG(ERROR) << "sandbox failed to start: " << result->ToString();
```

Comment:
Are things logged to stderr readable from the Javascript? I'm slightly confused by the socket usage above.

Reply:
They are, but we don't read them. But this might be useful if we do run into errors, that they will be in our logs at least.
Basically, all we care about is the process return code for the sandbox. The logs are ignored, and all communication back to the parent parent process goes indirectly via the socket.

Reply:
Fair enough, if we need to add errors based off of the sandbox2 final result code we can do that after